flycheck.el

```diff
@@ -4095,19 +4140,29 @@ refresh the mode line."
 
 STATUS defaults to `flycheck-last-status-change' if omitted or
 nil."
-  (let ((text (pcase (or status flycheck-last-status-change)
-                (`not-checked "")
-                (`no-checker "-")
-                (`running "*")
-                (`errored "!")
-                (`finished
-                 (let-alist (flycheck-count-errors flycheck-current-errors)
-                   (if (or .error .warning)
-                       (format ":%s|%s" (or .error 0) (or .warning 0))
-                     "")))
-                (`interrupted ".")
-                (`suspicious "?"))))
-    (concat " " flycheck-mode-line-prefix text)))
+  (let (text
+        face)
+    (setq text (concat " " flycheck-mode-line-prefix
+                       (pcase (or status flycheck-last-status-change)
+                         (`not-checked "")
+                         (`no-checker "-")
+                         (`running "*")
+                         (`errored (progn
+                                     (setq face 'error)
+                                     "!"))
+                         (`finished
+                          (let-alist (flycheck-count-errors flycheck-current-errors)
+                            (if (or .error .warning)
+                                (progn
+                                  (setq face 'error)
+                                  (format ":%s|%s" (or .error 0) (or .warning 0)))
+                              (setq face 'success)
+                              ":0")))
```

Comment:
Btw, why don't we make this something configurable so people can tweak it if they want to? E.g. `flycheck-mode-success-indicator` (a string). I think that'd be best as some people like like `0`, some `0:0`, and some might want something like a checkmark symbol or an "OK".

Reply:
Added `flycheck-mode-success-indicator'